framework/vectorstore/redis.go

```diff
@@ -0,0 +1,880 @@
+		if err != nil {
+			// If pipeline fails, mark all in this batch as failed
+			for _, id := range batch {
+				deleteResults = append(deleteResults, DeleteResult{
+					ID:     id,
+					Status: DeleteStatusError,
+					Error:  fmt.Sprintf("pipeline execution failed: %v", err),
+				})
+			}
+			continue
+		}
+
+		// Process results for this batch
+		for j, cmd := range cmds {
+			id := batch[j]
+			if cmd.Err() != nil {
+				deleteResults = append(deleteResults, DeleteResult{
+					ID:     id,
+					Status: DeleteStatusError,
+					Error:  cmd.Err().Error(),
+				})
+			} else if cmd.Val() > 0 {
+				// Key existed and was deleted
+				deleteResults = append(deleteResults, DeleteResult{
+					ID:     id,
+					Status: DeleteStatusSuccess,
+				})
+			} else {
+				// Key didn't exist
+				deleteResults = append(deleteResults, DeleteResult{
+					ID:     id,
+					Status: DeleteStatusError,
+					Error:  "document not found",
+				})
+			}
+		}
+	}
+
+	// If there are more results, continue with next cursor
+	if nextCursor != nil {
+		nextResults, err := s.deleteAllWithCursor(ctx, namespace, queries, nextCursor)
+		if err != nil {
+			return nil, fmt.Errorf("failed to delete remaining documents: %w", err)
+		}
+		// Combine results from this batch and subsequent batches
+		deleteResults = append(deleteResults, nextResults...)
+	}
+
+	return deleteResults, nil
+}
+
+func (s *RedisStore) DeleteNamespace(ctx context.Context, namespace string) error {
+	ctx, cancel := withTimeout(ctx, s.config.ContextTimeout)
+	defer cancel()
+
+	// First, delete all keys with the namespace prefix
+	pattern := fmt.Sprintf("%s:*", namespace)
+
+	// Use SCAN to find all keys with the namespace prefix
+	var cursor uint64 = 0
+	var keysToDelete []string
+
+	for {
+		keys, newCursor, err := s.client.Scan(ctx, cursor, pattern, 1000).Result()
+		if err != nil {
+			return fmt.Errorf("failed to scan keys for namespace %s: %w", namespace, err)
+		}
+
+		keysToDelete = append(keysToDelete, keys...)
+		cursor = newCursor
+
+		if cursor == 0 {
+			break
+		}
+	}
+
+	// Delete all keys in batches
+	if len(keysToDelete) > 0 {
+		batchSize := 100
+		for i := 0; i < len(keysToDelete); i += batchSize {
+			end := i + batchSize
+			if end > len(keysToDelete) {
+				end = len(keysToDelete)
+			}
+			batch := keysToDelete[i:end]
+
+			if err := s.client.Del(ctx, batch...).Err(); err != nil {
+				return fmt.Errorf("failed to delete keys for namespace %s: %w", namespace, err)
+			}
+		}
+	}
+
+	// Drop the index using FT.DROPINDEX
+	if err := s.client.Do(ctx, "FT.DROPINDEX", namespace).Err(); err != nil {
+		// Check if error is "Unknown Index name" - that's OK, index doesn't exist
+		if strings.Contains(err.Error(), "Unknown Index name") {
+			return nil // Index doesn't exist, nothing to drop
+		}
+		return fmt.Errorf("failed to drop semantic index %s: %w", namespace, err)
+	}
+
+	return nil
+}
+
+func (s *RedisStore) Close(ctx context.Context, namespace string) error {
```

Comment:
nope, just close the redis connection - or return nil if redis closure is automatically handled